src/lib/initMetric.ts

```diff
@@ -39,12 +42,15 @@ export const initMetric = (name: Metric['name'], value?: number): Metric => {
     }
   }
 
+  // Use `entries` type specific for the metric.
+  const entries: Extract<Metric, {name: MetricName}>['entries'] = [];
+
   return {
     name,
     value: typeof value === 'undefined' ? -1 : value,
-    rating: 'good', // Will be updated if the value changes.
+    rating: 'good' as const, // Will be updated if the value changes.
     delta: 0,
```

Comment:
Why do we need this `as const`? And how can it be a `const` if the value changes?

Reply:
> Why do we need this `as const`? And how can it be a `const` if the value changes?

It's kind of a misnomer in this case. Practically speaking, it's telling the compiler not to widen the type of `rating` to `string` and keep it as the literal `'good'` (since `rating` is typed to `'good' | 'needs-improvement' | 'poor'`, the type has to be one (or more) of those three, and not the more general `string`).

One way of thinking about it is that the _value_ `'good'` is being treated as `const`, but assigned to the property `rating`, which is not `const` and can be changed freely after this.

The other common approach besides `as const` would be `rating: 'good' as 'good'`, but
- that's slightly less idiomatic these days (`as const` was added to remove awkward self assertions, though usually uglier ones than this one)
- it's currently disallowed by the `prefer-as-const` lint rule.